src/datasources/accounts/counterfactual-safes/counterfactual-safes.datasource.spec.ts

```diff
@@ -0,0 +1,430 @@
+    await migrator.migrate();
+    mockConfigurationService.getOrThrow.mockImplementation((key) => {
+      if (key === 'expirationTimeInSeconds.default') return faker.number.int();
+    });
+
+    target = new CounterfactualSafesDatasource(
+      fakeCacheService,
+      sql,
+      mockLoggingService,
+      mockConfigurationService,
+    );
+  });
+
+  afterEach(async () => {
+    await sql`TRUNCATE TABLE accounts, account_data_settings, counterfactual_safes CASCADE`;
+    fakeCacheService.clear();
+    jest.clearAllMocks();
+  });
+
+  afterAll(async () => {
+    await testDbFactory.destroyTestDatabase(sql);
+  });
+
+  describe('createCounterfactualSafe', () => {
+    it('should create a Counterfactual Safe', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const [account] = await sql<
+        Account[]
+      >`INSERT INTO accounts (address) VALUES (${address}) RETURNING *`;
+      const createCounterfactualSafeDto =
+        createCounterfactualSafeDtoBuilder().build();
+
+      const actual = await target.createCounterfactualSafe(
+        account,
+        createCounterfactualSafeDto,
+      );
+      expect(actual).toStrictEqual(
+        expect.objectContaining({
+          id: expect.any(Number),
+          chain_id: createCounterfactualSafeDto.chainId,
+          creator: account.address,
+          fallback_handler: createCounterfactualSafeDto.fallbackHandler,
+          owners: createCounterfactualSafeDto.owners,
+          predicted_address: createCounterfactualSafeDto.predictedAddress,
+          salt_nonce: createCounterfactualSafeDto.saltNonce,
+          singleton_address: createCounterfactualSafeDto.singletonAddress,
+          threshold: createCounterfactualSafeDto.threshold,
+          account_id: account.id,
+        }),
+      );
+    });
+
+    it('should delete the cache for the account Counterfactual Safes', async () => {
+      const address = getAddress(faker.finance.ethereumAddress());
+      const [account] = await sql<
+        Account[]
+      >`INSERT INTO accounts (address) VALUES (${address}) RETURNING *`;
+      await target.createCounterfactualSafe(
+        account,
+        createCounterfactualSafeDtoBuilder().build(),
+      );
+      await target.getCounterfactualSafesForAccount(account);
+      const cacheDir = new CacheDir(`counterfactual_safes_${address}`, '');
+      await fakeCacheService.set(
+        cacheDir,
+        JSON.stringify([]),
+        faker.number.int(),
+      );
+
+      // the cache is cleared after creating a new CF Safe for the same account
+      await target.createCounterfactualSafe(
+        account,
+        createCounterfactualSafeDtoBuilder().build(),
+      );
+      expect(await fakeCacheService.get(cacheDir)).toBeUndefined();
+    });
+
+    // TODO: move this check to the repository.
+    it.skip('should throw if the counterfactual-safes storage setting is disabled', async () => {
```

Comment:
Is this intentionally skipped?

Reply:
Yes, this was intended to be moved to the repository, which will be added in a forthcoming PR. But giving this a second thought, I've just removed it from this PR as it's out of context. I will include it in the repository PR.

Reply:
Removed in [b6ae6a7](https://github.com/safe-global/safe-client-gateway/pull/1773/commits/b6ae6a7ef4f73295047cce15a4a36397944419ea)